api/dashboard/read

```diff
@@ -117,31 +117,35 @@ if ($cmd eq 'status') {
 
     my $recent = get_stats('5min');
     my $cpu = $recent->{'cpu_usage'} || 0;
-    chomp $cpu; # remove trailing '%' char
+    chop $cpu; # remove trailing '%' char
     $ret->{'proxy'}{'cpu_usage'} = sprintf('%.2f',$cpu);
 
 
     my $cdb = esmith::ConfigDB->open_ro('contentfilter');
-    $ret->{'filter'}{'profiles'} = scalar $cdb->get_all();
+    $ret->{'filter'}{'profiles'} = defined($cdb) ? scalar $cdb->get_all() : 0;
     $ret->{'filter'}{'status'} = $db->get_prop('ufdb', 'status');
     $ret->{'filter'}{'antivirus'} = $db->get_prop('squidclamav', 'status');
 
     # find latest modified signature
     my $av_db = -1;
-    opendir(DIR, "/var/lib/squidclamav");
-    while (my $file = readdir(DIR)) {
-        next if ($file =~ m/^\./);
-        my $fname = "/var/lib/squidclamav/$file";
+    my $av_dir_exists = opendir(DIR, "/var/lib/squidclamav");
 
-        if (-l $fname) {
-            $fname = readlink($fname);
-        }
-        my $timestamp = (stat($fname))[9];
-        if ($timestamp > $av_db) {
-            $av_db = $timestamp;
+    if (defined($av_dir_exists)) {
```

Comment:
Suggestion, simpler to understand:
```
if (-d "/var/lib/squidclamav") {
    opendir(DIR, "/var/lib/squidclamav");
```
See also: https://perldoc.perl.org/functions/-X.html